core/src/main/scala/cromwell/core/simpleton/WomValueBuilder.scala

```diff
@@ -92,7 +92,9 @@ object WomValueBuilder {
     }
 
     outputType match {
-      case _: WomPrimitiveType => components collectFirst { case SimpletonComponent(_, v) => v } get
+      case _: WomPrimitiveType =>
+        assert(components.size == 1)
```

Comment:
I think assert can take a second argument that describes the failure. Could be nice to add some explanation.

Reply:
why are we doing `assert`s?

Reply:
ToL: IMHO this and all others usages in Cromwell should be `Predef.require` instead of `Predef.assert`.

I don't like `Predef.assert` for the same reasons as java's `assert` keyword:
1. They have different meanings depending on how the program is [run](https://stackoverflow.com/questions/2758224/what-does-the-java-assert-keyword-do-and-when-should-it-be-used) or [compiled](http://daily-scala.blogspot.com/2010/04/elidable-remove-method-calls-at-compile.html). `require` _always_ throws an exception on bad inputs. `assert` only _sometimes_ throws an error on bad inputs.
2. When an `Predef.assert` or a java `assert` does trip an [`Error`](https://docs.oracle.com/javase/8/docs/api/java/lang/AssertionError.html) it indicates a "[serious problems that a reasonable application should not try to catch](https://docs.oracle.com/javase/8/docs/api/java/lang/Error.html)", often shutting down those reasonable applications. Again, IMHO, instead of Cromwell not being able to run at all, an `Exception` via `Predef.require` should be thrown instead where the one workflow will fail and the others will keep running until the bug can be reported and fixed.

Reply:
This is sort of why I like asserts. They get run with your tests so you know they're getting validated, but aren't part of the final code.

They're like comments saying "this should never happen" that actually get validated during tests.

Reply:
I hate asserts in real code

Reply:
IMO they're strictly better than comments like `"I know Condition X will always be true here"`

Reply:
Also true, but that doesn’t make assert non-sucky

Reply:
I think they can be a useful tool, but will agree that they do have with a *very* narrow use case. 

I'm happy to remove them if I'm the only one in Team "asserts are ok sometimes" (I added them for my own sanity checking when running the tests, I thought others might enjoy similar sanity verification at a later date)

Reply:
OK, vote here via "reaction emoji" on this comment:
- thumbs up: keep the `assert`s
- thumbs down: remove
- grin: "I'm just observing this"
- party: "I'm not even observing this"
- confused: "what's happening"
- heart: "please don't take my vote against you as a personal sleight"